sched/scheduler/stateful_scheduler.go

```diff
@@ -99,25 +100,39 @@ type jobAddedMsg struct {
 	saga *saga.Saga
 }
 
-func (s *statefulScheduler) ScheduleJob(job sched.Job) error {
-
+func (s *statefulScheduler) ScheduleJob(jobDef sched.JobDefinition) (string, error) {
 	defer s.stats.Latency("schedJobLatency_ms").Time().Stop()
 	s.stats.Counter("schedJobRequests").Inc(1)
 
+	job := sched.Job{
+		Id:  generateJobId(),
+		Def: jobDef,
+	}
+
 	// Log StartSaga Message
 	// TODO: need to serialize job into binary and pass in here
 	// so we can recover the job in case of failure
 	sagaObj, err := s.sagaCoord.MakeSaga(job.Id, nil)
 	if err != nil {
-		return err
+		return "", err
 	}
 
 	s.addJobCh <- jobAddedMsg{
 		job:  job,
 		saga: sagaObj,
 	}
 
-	return nil
+	return job.Id, nil
+}
+
+// generates a jobId using a random uuid
+func generateJobId() string {
+	id, err := uuid.NewV4()
+	for err != nil {
```

Comment:
I must have missed this before, but why would it return a transient error?

Also, I'd consider just doing:

for {
  if id, err := uuid.NewV4(); err == nil {
    return id.String()
  }
}